ch_pipeline/core/telescope.py

```diff
@@ -554,27 +559,44 @@ def beam(self, feed, freq_id):
             raise ValueError("The requested feed doesn't seem to exist.")
 
         if tools.is_array_x(feed_obj):
-            fname = self.primary_beamx_filename
-
+            pol_ind = list(self._primary_beam.pol[:]).index(b"x")
         elif tools.is_array_y(feed_obj):
-            fname = self.primary_beamy_filename
-
+            pol_ind = list(self._primary_beam.pol[:]).index(b"y")
```

Comment:
@tristpinsm can you copy out the various ancillary datasets (i.e. freq and pol) into memory during the `_finalise_config`? Depending on how well HDF5 caches the datasets (and I think not well at all unless the dataset is chunked) this may end up re-reading them every time it needs to use them. For pol, you can even precompute the pol_ind's in there.

Reply:
ok, sounds good. I'll leave the bytestring in there for now, but I'll make and issue in `caput` and link to this code so we don't forget.